capa/main.py

```diff
@@ -325,6 +339,15 @@ def pbar(s, *args, **kwargs):
     return matches, meta
 
 
+def find_capabilities(ruleset: RuleSet, extractor: FeatureExtractor, **kwargs) -> Tuple[MatchResults, Any]:
+    if isinstance(extractor, StaticFeatureExtractor):
+        extractor_: StaticFeatureExtractor = cast(StaticFeatureExtractor, extractor)
+        return find_capabilities_static(ruleset, extractor_, kwargs)
+    else:
+        # extractor_ = cast(DynamicFeatureExtractor, extractor)
+        print("nni")
```

Comment:
as above, add explicit `elif` case and then raise `NotImplementedError` if its a dynamic extractor.

Reply:
tbh, im not sure at what level we'll want to split the static and dynamic flavor logic. it might make sense here, as you've sketched. but it might also make sense at a higher level, since we won't mix the static/dynamic results with static/dynamic rendering, so duplicating the type switches might be bad.

no need to change now, we're still exploring and sketching. but something to keep in mind.

Reply:
i feel like one upside of doing the split here is that it would be easier to support running a sample against several extractors in the future. we would make `get_extractor()` yield several extractors,  pass each one to this function, and then render the results of each output. but as you said, it's probably early to think of that.